aten/src/ATen/Dispatch.h

```diff
@@ -8,7 +8,7 @@
 
 #define AT_PRIVATE_CASE_TYPE(enum_type, type, ...) \
   case enum_type: {                                \
-    using scalar_t = type;                         \
+    using scalar_t C10_UNUSED = type;              \
```

Comment:
What is the usage of this? I think I'm missing sth

Reply:
Dispatch macro sets `scalar_t` with `using scalar_t C10_UNUSED = type; `. This may or may not be used in the offloading function. This results in compiler warning about unused typedef. Adding `C10_UNUSED` adds a compile-time attribute to ignore that warning.

Reply:
Ok! Understand!